### ISSUE TITLE
running a fisrt sketch

### ISSUE THREAD

**Issue by ror6ax** [quil/quil#211]
Hi, just starting out in Clojure and Quil here.

Running `lein run -m hello-quil.core ` gives me the error Cannot run sketch without a display. Read this for possible solutions: https://github.com/processing/processing/wiki/Running-without-a-Display

So I did, but running `xvfb-run lein run -m hello-quil.core` gives the same result. 

Please advice how can I resolve this issue.

Thanks.

**Comment by nbeloglazov** [comment]
Hi Gregory. Where are you running quil?

**Comment by ror6ax** [comment]
Fedora 25, my laptop. I'm surprised since it looks like quil does not recognize I run graphic environment...

**Comment by ror6ax** [comment]
Is there a way to render directly into file? `-o filename.png`  kind of thing.

**Comment by ror6ax** [comment]
```
(ns core
  (:use quil.core))


(defn setup [])


(defn draw []
  (background 153)
  (line 0 0 300 300)
  (exit))


(defsketch circles
  :title "Circles"
  :setup setup

  :renderer :pdf
  :output-file "./tt.pdf"

  :draw draw
  :size [300 300])

(defn -main [] ())
```

Tried :output-file. Same result - error, no pdf anywhere.

**Comment by nbeloglazov** [comment]
Sorry for probably very late response. I just tried. 

1. Rent a headless server on google compute engine.
2. Install openjdk 8, leiningen, other things from the [article](https://github.com/processing/processing/wiki/Running-without-a-Display)
3. Create project using quil lein template: `lein new quil test`.
4. Add `(q/save-frame "foo-####.png)` to the `src/test/core.clj` at the end of `draw-state`. 
5. Run `xvfb-run lein run -m test.core`. 
6. Checked and I could see png files in the directory. 

So it worked for me. How did you run xvfb-run?
